solc/CommandLineParser.cpp

```diff
@@ -189,15 +189,13 @@ void CommandLineParser::printVersionAndExit()
 		"Version: " <<
 		solidity::frontend::VersionString <<
 		endl;
-	exit(EXIT_SUCCESS);
```

Comment:
It's not a simple as just removing the `exit()` :) The compiler must still exit after printing this info, just not like this. You should instead make `CommandLineParser::parse` return `false` after it runs one of these functions. But now returning `false` means that there was an error - which is why you first need to ensure that returning with an error is only done by raising an exception.